StringOperations.java

```diff
@@ -48,84 +53,63 @@ public static void main(String[] args) {
-                                    match = false;
-                                    break;
-                                }
-                            }
-                            if (match) {
-                                replaceResult.append(newSubstring);
-                                i += oldSubstring.length() - 1;
-                            } else {
-                                replaceResult.append(inputString.charAt(i));
-                            }
-                        }
-                        resultString = replaceResult.toString();
+                        resultString = inputString.replace(oldSubstring, newSubstring);
                         System.out.println("Updated String: " + resultString);
                     } else {
                         System.out.println("Invalid input: Both the old and new substrings must be non-empty.");
                     }
                     break;
-                case 4:// is palindrome 
-                    boolean isPal = true;
-                    int n = inputString.length();
-                    for (int i = 0; i < n / 2; i++) {
-                        if (inputString.charAt(i) != inputString.charAt(n - i - 1)) {
-                            isPal = false;
-                            break;
-                        }
-                    }
+
+                case 4: // Is Palindrome
+                    boolean isPal = isPalindrome(inputString);
                     System.out.println("Is palindrome: " + isPal);
                     break;
-                case 5:// splice operation
-                    System.out.print("Enter the start index for splice: ");
-                    if (scanner.hasNextInt()) {
-                        int start = scanner.nextInt();
-                        System.out.print("Enter the length for splice: ");
-                        if (scanner.hasNextInt()) {
-                            int length = scanner.nextInt();
-                            scanner.nextLine(); 
-                            if (start >= 0 && start + length <= inputString.length()) {
-                                resultString = inputString.substring(0, start) + inputString.substring(start + length);
-                                System.out.println("Updated String: " + resultString);
-                            } else {
-                                System.out.println("Invalid input: Start index and length are out of bounds.");
-                            }
-                        } else {
-                            System.out.println("Invalid input: Length must be an integer.");
-                            scanner.next(); 
-                        }
-                    } else {
-                        System.out.println("Invalid input: Start index must be an integer.");
-                        scanner.next(); 
-                    }
+
+                case 5: // Splice
```

Comment:
Ensure uniform spacing between cases for readability. There should be a blank line between break; and the next case.